gilbreth_perception/src/recognition_node.cpp

```diff
@@ -384,6 +387,22 @@ class RecognitionClass {
       pick_point.z = pick_pose[result.item_id][2];
       pick_point_cloud->push_back(pick_point);
       pcl::transformPointCloud(*pick_point_cloud, *rotated_pick_point_cloud, result.final_transformation);
+      // Use ICP to fine align model to scene
+      t_start = std::clock();
+      pcl::IterativeClosestPoint<pcl::PointXYZ, pcl::PointXYZ> icp;
+      icp.setMaximumIterations (iterations);
+      icp.setInputSource(rotated_model);
+      icp.setInputTarget(scene);
+      pcl::PointCloud<pcl::PointXYZ> Final;
+      icp.align(Final);
+      if (print_detailed_info){
+    	duration = (std::clock() - t_start) / (double)CLOCKS_PER_SEC;
+    	std::cerr << "align time is " << duration << " seconds." << std::endl;
```

Comment:
Try using the ROS_INFO, ROS_ERROR, etc.  macros for printing to the console. You can use the ROS_INFO_STREAM and ROS_ERROR_STREAM if you'd like to use the **<<** stream insertion operators.

Reply:
Will do that.